docs-standalone/coding-style-guide.md

```diff
@@ -0,0 +1,299 @@
+
+    // DO
+    [myObject doFooWith:arg1 name:arg2 error:arg3];
+
+    [myObject doFooWith:arg1
+                   name:arg2
+                  error:arg3];
+
+    // DON'T
+    [myObject doFooWith:arg1 name:arg2
+                  error:arg3];
+
+    [myObject doFooWith:arg1
+              name:arg2
+              error:arg3];
+
+#### 1.4. Appledoc Header Comments
+
+*ResearchKit* uses [appledoc](http://appledoc.gentlebytes.com/appledoc/) to generate its documentation from specially marked comments in header files.
+
+Follow these guidelines when writing *appledoc comments*:
+
+- Multiline *appledoc comments* start with the `/**` character sequence.
+- When documenting methods, use the `@param` and `@return` keywords to detail the parameters and return value.
+- When you name classes or methods, enclose them in backticks so *appledoc* creates a reference (`` `ORKStep` is ...``).
+- For multiline code examples, surround them with a triple backtick (```) for cross references within the code block not to be automatically generated.
+- Don't use abbreviations such as *e.g.* or *i.e.* in the documentation.
+- Hard wrap comment lines at column 100.
+- Read the latest *ResearchKit* documentation for inspiration and try to follow the same literary style.
+
+#### 1.5. Newlines
+
+Use exactly two empty lines to separate:
+- The `/* Copyright header */` and the `#import` section.
+- The `#import` section and the class `@interface` or `@implementation` line (or its associated *forward declarations* or *appledoc comment*).
+- Different `@interface` or `@implementation` sections within the same file.
+
+Do not use two or more empty lines in any other cases.
+
+Use exactly one empty line to separate:
+- The last *forward declaration* and the *`@interface` declaration*.
+- The *`@interface` declaration* and the first *method* or *property declaration*.
+- The last *method* or *property declaration* and the *`@end` keyword*.
+- The *`@implementation` line* and the first *method definition*.
+- The closing bracket of the last *method definition* and the *`@end` keyword*.
+- The *`@param` section* and the *`@return` line* in an *appledoc comment*.
+
+*Header* and *implementation* files must have one, and only one, trailing empty line.
+
+Do not use empty lines to separate:
+- *Forward declarations* from other contiguous *forward declarations*.
+- An *appledoc comment* from its related *class* or *method*.
+- Contiguous *`@param` lines* within the same *appledoc comment*.
+- Last statement in a method definition and its closing bracket.
+
+Note that *forward declarations* should appear before any *class appledoc comment*.
+
+You can optionally use one (and only one) blank like to separate:
+- Groups of related `#import` statements.
+- Groups of related statements in a single method implementation.
+
+
+#### 1.6. Header File Example
+
+    /*
+     Copyright (c) 2015, Apple Inc. All rights reserved.
+     ...
```

Comment:
How about replacing `Apple Inc.` with `AUTHOR'S NAME` to remind developer to put his/her name in the copyright line when creating a new file.